corehq/blobs/fsdb.py

```diff
@@ -75,20 +88,29 @@ def delete(self, name=None, bucket=DEFAULT_BUCKET):
             path = safejoin(self.rootdir, bucket)
             remove = shutil.rmtree
         else:
-            path = self._getpath(name, bucket)
+            path = self.get_path(name, bucket)
             remove = os.remove
         if not exists(path):
             return False
         remove(path)
         return True
 
-    def _getpath(self, name, bucket):
+    @staticmethod
+    def get_unique_name(basename):
+        if SAFENAME.match(basename) and "/" not in basename:
+            prefix = basename
+        else:
+            prefix = "unsafe"
+        return prefix + "." + uuid4().hex
```

Comment:
Oh, that's a great idea. I'm starting to wonder if the basename thing is worth the extra complexity it adds. I could easily remove it now that each blob has a unique id.